engine/swarm/secrets.md

```diff
@@ -0,0 +1,339 @@
+    managers or workers to this swarm, they will receive the key used to encrypt
+    and decrypt the secrets over a TLS-encrypted connection.
+
+2.  Create a new overlay network `wp_network` which the MySQL and WordPress
+    services will use to communicate.
+    ```bash
+    $ docker network create -d overlay wp_network
+    ```
+
+3.  Generate a random alphanumeric password for MySQL and store it as a Docker
+    secret with the name `mysql_password` using the `docker secret create`
+    command. To make the password shorter or longer, adjust the last argument of
+    the `openssl` command. This is just one way to create a relatively random
+    password. You can use another command to generate the password if you choose.
+    Notice that the password is never exposed on the command line or saved to a
+    plain-text file.
+
+    >**Note**: After you create a secret, you cannot update it. You can only
+    remove and re-create it, and you cannot remove a secret that a service is
+    using. However, you can grant or revoke a running service's access to
+    secrets using `service update`. If you need the ability to update a secret,
+    consider adding a version component to the secret name, so that you can later
+    add a new version, update the service to use it, then remove the old version.
+
+    ```bash
+    $ openssl rand -base64 20 |docker secret create mysql_password
+      lwxgks2rojdbvaui5kay1suvr
+    ```
+
+    The value returned is not the password, but the ID of the secret.
+    List the secrets managed by Docker using `docker secret ls`:
+
+    ```bash
+    $ docker secret ls
+    ID                          NAME                CREATED             UPDATED
+    lwxgks2rojdbvaui5kay1suvr   mysql_password      41 seconds ago      41 seconds ago
+    ```
+    The secret is stored in the encrypted RAFT logs for the swarm.
+
+4.  Create the MySQL service. The `mysql` image now supports additional
+    environment variables which allows the container to read credentials and other
+    important settings from files within the container. This example uses the new
+    `MYSQL_ROOT_PASSWORD_FILE` variable and sets it to the file path within the
+    container which contains the `mysql_password` secret.
+
+    >**Note**: To modify your own containers to support this feature, see
+    [Build support for Docker Secrets into your images](store_sensitive_strings.md#build-support-for-docker-secrets-into-your-images).
+
+    The MySQL service will have the following characteristics.
+
+    - Joins the service to the `wp_network` overlay network, and resolves to the
+      DNS name `mysql`.
+    - Because the scale is set to 1, only a single MySQL task runs.
+    - Grants the service access to the `mysql_password` secret, which will be
+      available as plain-text within the container at `/run/secrets/mysql_password`.
+    - Sets the environment variable `MYSQL_ROOT_PASSWORD_FILE` to this file path.
```

Comment:
I think we should follow best practice here and don't use `MYSQL_ROOT_PASSWORD` for the database connection, but specify a `MYSQL_DATABASE `, `MYSQL_USER`, and `MYSQL_PASSWORD` so that a non-database-root user is used for the WordPress connection / database